Conceptpower+Spring/src/main/java/edu/asu/conceptpower/app/core/impl/AlternativeIdService.java

```diff
@@ -0,0 +1,50 @@
+package edu.asu.conceptpower.app.core.impl;
+
+import java.util.Collection;
+import java.util.HashSet;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import edu.asu.conceptpower.app.core.ConceptTypesService;
+import edu.asu.conceptpower.app.core.ConceptTypesService.ConceptTypes;
+import edu.asu.conceptpower.core.ConceptEntry;
+
+@Service
+public class AlternativeIdService {
+
+    @Autowired
+    private ConceptTypesService conceptTypesService;
+
+    public void addAlternativeIds(String queriedId, ConceptEntry entry) {
+        if (entry.getAlternativeIds() == null) {
+            entry.setAlternativeIds(new HashSet<String>());
+        }
+        if (conceptTypesService.getConceptTypeByConceptId(queriedId) == ConceptTypes.GENERIC_WORDNET_CONCEPT) {
+            entry.getAlternativeIds().add(queriedId);
+        }
+        if (conceptTypesService.getConceptTypeByConceptId(queriedId) != ConceptTypes.SPECIFIC_WORDNET_CONCEPT) {
+            // Added for generic wordnet and specific local concept
+            entry.getAlternativeIds().add(entry.getId());
+        }
+        // Specific Wordnet id is added irrespective of what is queried for
+        if (entry.getWordnetId() != null) {
+            String[] wordNetIds = entry.getWordnetId().split(",");
+            for (String wordNetId : wordNetIds) {
+                entry.getAlternativeIds().add(wordNetId);
+            }
+        }
+    }
```

Comment:
alternative ids should include all ids, shouldn't it? So ConceptId, WordnetId, and if general wordnet id was queried for that as well. Right now, you're not adding the concept id, do you? And what about merged ids?